### PR TITLE
scripts: tests: Blackbox test expansion

### PR DESCRIPTION
**Pull request by LukaszMrugala**
Adds five new tests, covering six more flags:
* `--test`, `-s`
* `--sub-test`
* `--integration`, `-G`
* `--test-config`
* `--level`
* `--filter`

### CODE REVIEW

File: scripts/tests/twister_blackbox/test_error.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        pass
+
+    @pytest.mark.usefixtures("clear_log")
+    @pytest.mark.parametrize(
+        'test, expected_exception',
+        TESTDATA_1,
+        ids=['valid', 'invalid']
+    )
+    @mock.patch.object(TestPlan, 'TESTSUITE_FILENAME', testsuite_filename_mock)
+    def test_test(self, out_path, test, expected_exception):
```
Comment (hakehuang) [inline scripts/tests/twister_blackbox/test_error.py:49]: can we add fucntion doc string here to explain  details of test approach? it is not friendly to see code to undersand the logic.

Comment (LukaszMrugala) [reply scripts/tests/twister_blackbox/test_error.py:49]: I think that such a comment shouldn't be in a specific test_ file, as it is information pertinent to any blackbox test. Where should be put such information then? In conftest? In a test_dummy.py, which would hold a single, highly commented testcase?

Comment (golowanow) [reply scripts/tests/twister_blackbox/test_error.py:49]: > I think that such a comment shouldn't be in a specific test_ file, as it is information pertinent to any blackbox test. Where should be put such information then? In conftest? In a test_dummy.py, which would hold a single, highly commented testcase?

`./scripts/tests/twister_blackbox/README.rst` ?

Comment (LukaszMrugala) [reply scripts/tests/twister_blackbox/test_error.py:49]: I will create a separate PR for a readme file.

**Review by golowanow — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**